core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2262,21 +2262,39 @@ private[spark] object Utils extends Logging {
   }
 
   /**
-   * Return whether dynamic allocation is enabled in the given conf
-   * Dynamic allocation and explicitly setting the number of executors are inherently
-   * incompatible. In environments where dynamic allocation is turned on by default,
-   * the latter should override the former (SPARK-9092).
+   * Return whether dynamic allocation is enabled in the given conf.
    */
   def isDynamicAllocationEnabled(conf: SparkConf): Boolean = {
-    val numExecutor = conf.getInt("spark.executor.instances", 0)
     val dynamicAllocationEnabled = conf.getBoolean("spark.dynamicAllocation.enabled", false)
-    if (numExecutor != 0 && dynamicAllocationEnabled) {
-      logWarning("Dynamic Allocation and num executors both set, thus dynamic allocation disabled.")
-    }
-    numExecutor == 0 && dynamicAllocationEnabled &&
+    dynamicAllocationEnabled &&
       (!isLocalMaster(conf) || conf.getBoolean("spark.dynamicAllocation.testing", false))
   }
 
+  /**
+   * Return the minimum number of executors for dynamic allocation.
+   */
+  def getDynamicAllocationMinExecutors(conf: SparkConf): Int = {
+    conf.getInt("spark.dynamicAllocation.minExecutors", 0)
```

Comment:
What's the value of these constants over util methods? These constants live in the yarn module, but there are uses in the `ExecutorAllocationManager` in core. Rather than having the constants in two places, with possibly different defaults, I thought a util method was appropriate.